packages/vaadin-overlay/src/vaadin-overlay.js

```diff
@@ -771,10 +771,23 @@ class OverlayElement extends ThemableMixin(DirMixin(ControllerMixin(PolymerEleme
     if (document.body.style.pointerEvents !== 'none') {
       // Set body pointer-events to 'none' to disable mouse interactions with
       // other document nodes.
-      this._previousDocumentPointerEvents = document.body.style.pointerEvents;
+      this._previousBodyPointerEvents = document.body.style.pointerEvents;
       document.body.style.pointerEvents = 'none';
     }
 
+    const { documentElement } = document;
+
+    if (documentElement.style.overflow !== 'hidden') {
+      this._previousDocumentOverflow = documentElement.style.overflow;
+      documentElement.style.overflow = 'hidden';
+    }
+
+    const isDocumentElementScrollable = documentElement.scrollHeight > document.clientHeight;
```

Comment:
**todo:** It is not a good idea to add `scrollbar-gutter` only when the container is scrollable because the scrollbar can be visible even when nothing to scroll.